components/eventing-controller/controllers/subscriptionv2/eventmesh/test/reconciler_integration_test.go

```diff
@@ -0,0 +1,457 @@
+			givenSinkFormat: "https://%s.%s.svc.cluster.local:8080%s",
+		},
+		{
+			name:            "should succeed to fix invalid sink with Url without port in subscription",
+			givenSinkFormat: "https://%s.%s.svc.cluster.local%s",
+		},
+	}
+
+	for _, testCase := range testCases {
+		tc := testCase
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			g := gomega.NewGomegaWithT(t)
+			ctx := context.Background()
+
+			// create unique namespace for this test run
+			testNamespace := getTestNamespace()
+			ensureNamespaceCreated(ctx, t, testNamespace)
+			subName := fmt.Sprintf("test-sink-%s", testNamespace)
+			sinkPath := "/path1"
+
+			// update namespace information in given test assets
+			givenSubscription := givenSubscriptionFunc(testNamespace, subName)
+			givenUpdateSubscription := givenUpdateSubscriptionFunc(testNamespace, subName, tc.givenSinkFormat, sinkPath)
+
+			// create a subscriber service
+			subscriberSvc := reconcilertesting.NewSubscriberSvc(subName, testNamespace)
+			ensureK8sResourceCreated(ctx, t, subscriberSvc)
+
+			// create subscription
+			ensureK8sResourceCreated(ctx, t, givenSubscription)
+			createdSubscription := givenSubscription.DeepCopy()
+			// check if the created subscription is correct
+			getSubscriptionAssert(ctx, g, createdSubscription).Should(wantSubscriptionMatchers)
+
+			// update subscription with valid sink
+			givenUpdateSubscription.ResourceVersion = createdSubscription.ResourceVersion
+			ensureK8sResourceUpdated(ctx, t, givenUpdateSubscription)
+
+			// check if an APIRule was created for the subscription
+			getAPIRuleForASvcAssert(ctx, g, subscriberSvc).Should(reconcilertestingv1.HaveNotEmptyAPIRule())
+
+			// check if the created APIRule is as required
+			apiRules, err := getAPIRulesList(ctx, subscriberSvc)
+			assert.NoError(t, err)
+			apiRuleUpdated := filterAPIRulesForASvc(apiRules, subscriberSvc)
+			getAPIRuleAssert(ctx, g, &apiRuleUpdated).Should(gomega.And(
+				reconcilertestingv1.HaveNotEmptyHost(),
+				reconcilertestingv1.HaveNotEmptyAPIRule(),
+				reconcilertestingv1.HaveAPIRuleSpecRules(acceptableMethods, object.OAuthHandlerName, sinkPath),
+				reconcilertestingv1.HaveAPIRuleOwnersRefs(givenSubscription.UID),
+			))
+
+			// update the status of the APIRule to ready (mocking APIGateway controller)
+			ensureAPIRuleStatusUpdatedWithStatusReady(ctx, t, &apiRuleUpdated)
+
+			// check if the updated subscription is correct
+			getSubscriptionAssert(ctx, g, givenSubscription).Should(wantUpdateSubscriptionMatchers)
+
+			// check if the reconciled subscription has API rule in the status
+			assert.Equal(t, givenSubscription.Status.Backend.APIRuleName, apiRuleUpdated.Name)
+
+			// check if the EventMesh mock received requests
+			_, postRequests, _ := countEventMeshRequests(
+				emTestEnsemble.nameMapper.MapSubscriptionName(givenSubscription.Name, givenSubscription.Namespace),
+				reconcilertesting.EventMeshOrderCreatedV1Type)
```

Comment:
I didn't really understand, whether it makes sense to count the requests to BEB in our tests. I think it also caused some flakiness. Isn't it sufficient to check that APIRule was updated ?

Reply:
Since we are using a mock for EventMesh, so we cannot actually send and receive a msg. IMO, we should make sure if the requests were made to create subscriptions on EventMesh server. And the test does not count exact number of requests, instead it only checks `GreaterOrEqual` than zero so it would not be flaky.

Reply:
Now I get it, I see the point now. We could maybe introduce a simple stub, which would just calculate the post/get requests. That could make the code easier. 

But you don't have to implement it in this PR or implement in the first place :)